flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/api/TableEnvironment.scala

```diff
@@ -111,9 +112,11 @@ abstract class TableEnvironment(val config: TableConfig) {
   private val externalCatalogs = new mutable.HashMap[String, ExternalCatalog]
 
   // temporary bridge between API and planner
-  private[flink] val expressionBridge: ExpressionBridge[PlannerExpression] =
+  protected val expressionBridge: ExpressionBridge[PlannerExpression] =
     new ExpressionBridge[PlannerExpression](functionCatalog, PlannerExpressionConverter.INSTANCE)
 
+  private[flink] val operationTreeBuilder = new OperationTreeBuilder(this)
```

Comment:
comment: we should definitely reduce the dependency of `OperationTreeBuilder` in the future. It should not be necessary to pass the entire `TableEnvironment` API class.

Reply:
Yes, totally agree. We will need only table lookup from `TableEnvironment`.